File: src/AnagramFinderBase.java

/**
 *
 */
public class AnagramFinderBase extends AnagramFinder {

	@Override
	public void findAnagrams(String[] words, String[] dict, int[] result) {

        for (int i = 0; i < words.length; i++) {
            String w = words[i];

            for (String a : dict) {

				if (w.length() == a.length() && isAnagram(w.toCharArray(), a.toCharArray()))
					result[i]++;
            }
        }
	}


	private boolean isAnagram(char[] word, char[] anagram) {

		boolean found;
		for (char w : word) {

			found = false;
			for (int j = 0; j < anagram.length && !found; j++) {
				if (w == anagram[j]) {
					anagram[j] = ' ';
					found = true;
				}
			}
			if (!found)
				return false;
		}

		for (char c : anagram)
			if (c != ' ') return false;

		return true;
	}
}

File: AnagramFinder/AnagramFinderBase.cs
using System;

namespace AnagramFinder
{
    public class AnagramFinderBase : IAnagramFinder {

        public void FindAnagrams(string[] words, string[] dict, int[] result) {

            for (int i = 0; i < words.Length; i++) {
                string w = words[i];

                foreach (string a in dict) {

                    if (w.Length == a.Length && isAnagram(w.ToCharArray(), a.ToCharArray()))
                        result[i]++;
                }
            }
        }


        private bool isAnagram(char[] word, char[] anagram) {

            bool found;
            foreach (char w in word) {

                found = false;
                for (int j = 0; j < anagram.Length && !found; j++) {
                    if (w == anagram[j]) {
                        anagram[j] = ' ';
                        found = true;
                    }
                }
                if (!found)
                    return false;
            }

            foreach (char c in anagram)
                if (c != ' ') return false;

            return true;
        }
    }
}
